### PR TITLE
`metadeploy_publish` consumes legal targets for supported orgs

### PR DESCRIPTION
**Pull request by Br4nd0R**
In the `cumulusci.yml` file, `plans` can now be configured with an `allowed_org_providers` attribute. Valid values are `user` and `devhub`. The `user` value means that the plan is intended users to provide their own orgs. The `devhub` value means that the plan can generate scratch orgs for product demos and/or testing. It is valid to list both values simultaneously.

### CODE REVIEW

File: cumulusci/tasks/metadeploy.py [hunk 1]
```diff
@@ -218,7 +224,27 @@ def _publish_plan(self, product, version, plan_name, plan_config, steps):
...
+        plan = Plan.parse_obj(self.project_config.config["plans"][plan_name])
+        return plan.allowed_org_providers
+
+    def _convert_org_providers_to_plan_equivalent(
+        self, providers: Union[None, List[str]]
+    ) -> str:
+        """Given a list of plan.allowed_org_providers return the value that
+        corresponds `supported_orgs` field on the Plan model in MetaDeploy
+        """
+        if not providers or providers == ["user"]:
```
Comment (Br4nd0R) [inline cumulusci/tasks/metadeploy.py:240]: Is this an okay default?

**Comment by davidmreed** [comment]
I like the terminology you chose here. It leaves the door open to a potential future implementation of the SignupRequest API to provision persistent orgs.

**Review by davidmreed — DISMISSED** [review]

File: cumulusci/utils/yaml/cumulusci_yml.py [hunk 2]
```diff
@@ -123,6 +123,7 @@ class Plan(CCIDictModel):  # MetaDeploy plans
     error_message: str = None
     post_install_message: str = None
     preflight_message: str = None
+    allowed_org_providers: List[Literal["devhub", "user"]] = None
```
[the remark this answers is not in the document]
Comment (Br4nd0R) [reply cumulusci/utils/yaml/cumulusci_yml.py]: I'll opt to set it to `["user"]`

**Review by jstvz — COMMENTED** [review]
I'm not convinced that using different values in CumulusCI and MetaDeploy is the best approach. `user` and `devhub` make sense to me, but using `scratch` and `persistent` on the MetaDeploy side makes less sense because users can _provide_ either scratch or persistent orgs.

Future enhancements to use the Signup API can use `signup`. 

(Thanks for cleaning up while you're in here.)

**Review by davidmreed — APPROVED** [review]

File: cumulusci/tasks/metadeploy.py [hunk 3]
```diff
@@ -24,7 +26,7 @@ def _init_task(self):
         metadeploy_service = self.project_config.keychain.get_service("metadeploy")
         self.base_url = metadeploy_service.url
         self.api = requests.Session()
-        self.api.headers["Authorization"] = "token {}".format(metadeploy_service.token)
+        self.api.headers["Authorization"] = f"token {metadeploy_service.token}"
```
[the remark this answers is not in the document]
Comment (Br4nd0R) [reply cumulusci/tasks/metadeploy.py:29]: 🧹

**Pull request merged**